packages/gatsby-plugin-google-analytics/src/gatsby-node.js

```diff
@@ -1,3 +1,50 @@
+// Copied from https://gist.github.com/faisalman/924970#file-analytics-regexp-js
+// NOTE: This is just an example, in many cases we'd want to async check some API
+async function isTrackingId(str) {
+  return /^ua-\d{4,9}-\d{1,4}$/i.test(str)
+}
+
+exports.pluginOptionsSchema = ({ Joi }) =>
+  Joi.object({
+    trackingId: Joi.string()
+      .required()
+      .description(
+        `The property ID; the tracking code won't be generated without it`
+      )
+      .external(async value => isTrackingId(value)),
```

Comment:
Should there, or can there be a distinction between structural validation and existence validation?

Ideally we'd want to check whether something is a valid email address format in all instances, and only check that the email address is active sometimes.

Reply:
Did some looking up here, you may have opted to make this external for example purposes. But ideally we'd want users to validate something with regex using [pattern](https://joi.dev/api/?v=17.2.1#stringpatternregex-name--options---aliases-regex). There is also email validation and some other "structure" based checks.

That would leave external to cover use cases that actually require external API calls, like the Contentful space check.

Reply:
Absolutely true!

I solely wanted to show what external validation would look like in the code, but couldn't find a Google API to validate a tracking ID, so I wrote this fake async validation function instead 😉